lib/cli/run.js

```diff
@@ -299,8 +299,8 @@ exports.builder = yargs =>
     .number(types.number)
     .alias(aliases);
 
-exports.handler = argv => {
+exports.handler = function(argv) {
   debug('post-yargs config', argv);
   const mocha = new Mocha(argv);
-  runMocha(mocha, argv);
+  return Promise.resolve(runMocha(mocha, argv));
```

Comment:
I have never liked to get an Yargs error, when some exception in eg the runner was thrown. IMO Yargs should not handle those type of exceptions. Yargs should parse and handle exceptions thrown during parsing, nothing more.

We could activate ECMA2018, turn this line into `async`/`await` and add `try`/`catch` . The error path would be shorter.

Reply:
I couldn't make it run in any other way. I really tried, but I believe I failed because it is actually `yargs` that is the top-level code that runs everything, so everything goes through it. But I have to admit that the code confused me, and maybe a second look now that I'm more familiar with the code.

Any idea on what I _should_ do?

Reply:
You could catch the error here:
```js
return Promise.resolve(runMocha(mocha, argv)).catch(function(err) {
  console.error(`\n${symbols.error} ${ansi.red('ERROR:')} ${err.message}`);
  process.exit(1);
});
```
There are two tests failing though, which probably could get fixed.
That way we would not have to touch the yargs part.

Reply:
I'll try to do this again.

Reply:
Done! It seems the problems stemmed from the weird sync/async trick I did, and once we go full async, there's no problem anymore and all error handling can be done from `run.js`.